### PR TITLE
Single client generation per schema

### LINKED ISSUE #92: Add single client generation for a single schema.
**Description:**
Currently we can generate multiple clients for a single schema by specifying multiple documents in the configuration. We have to improve this to create a single client per schema to comply with IDL support integration requirements.

### PR DESCRIPTION
**Pull request by udda1996**
## Purpose
Improve the client generator to create a single client file for one schema. 

## Goals
fixes #92

### CODE REVIEW

**Review by NipunaRanasinghe — COMMENTED** [review]
Also we need to update the existing GraphQL tool guide (https://ballerina.io/learn/graphql-client-tool/) along with these new changes

File: graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java [hunk 1]
```diff
@@ -100,19 +103,16 @@ public static ClientGenerator getInstance() {
     /**
      * Generates the client file content.
      *
-     * @param queryDocument                     the object instance of the queries document
-     * @param queryDocumentName                 the name of the queries document
      * @param graphQLSchema                     the object instance of the GraphQL schema (SDL)
```
Comment (RolandHewage) [inline graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java:106]: Let's add the documentation comment for the new param `queryDocuments`.

Suggestion `@param queryDocuments        the list of names of the queries documents`

File: graphql-cli/src/main/java/io/ballerina/graphql/cmd/GraphqlCmd.java [hunk 2]
```diff
@@ -121,6 +121,9 @@ public void execute() {
             Config config = readConfig();
             ConfigValidator.getInstance().validate(config);
             List<GraphqlProject> projects = populateProjects(config);
+//            if (projects.size() > 1) {
+//                AnnotationsValidator.getInstance().validate(config);
+//            }
```
Comment (RolandHewage) [inline graphql-cli/src/main/java/io/ballerina/graphql/cmd/GraphqlCmd.java:126]: Let's remove these comments.

File: graphql-cli/src/main/java/io/ballerina/graphql/generator/CodeGenerator.java [hunk 3]
```diff
@@ -119,16 +116,10 @@ private List<SrcFilePojo> generateBalSources(GraphqlProject project) throws Clie
...
-
-            String clientSrc = ClientGenerator.getInstance().
-                    generateSrc(queryDocument, queryDocumentName, schema, authConfig);
-            sourceFiles.add(new SrcFilePojo(SrcFilePojo.GenFileType.GEN_SRC, projectName,
-                    CodeGeneratorUtils.getClientFileName(documentFile), clientSrc));
-        }
+        String clientSrc = ClientGenerator.getInstance().
+                generateSrc(documents, schema, authConfig);
+        sourceFiles.add(new SrcFilePojo(SrcFilePojo.GenFileType.GEN_SRC, projectName,
+                "client.bal", clientSrc));
```
Comment (RolandHewage) [inline graphql-cli/src/main/java/io/ballerina/graphql/generator/CodeGenerator.java:122]: Let's define the client file name (i.e. client.bal) as a constant in the `CodeGeneratorConstants.java` file.

File: graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java [hunk 4]
```diff
@@ -126,13 +126,13 @@ public String generateSrc(Document queryDocument, String queryDocumentName, Grap
      * @param authConfig                the object instance representing authentication configuration information
      * @return                          Syntax tree for the ballerina client code
      */
-    private SyntaxTree generateSyntaxTree(Document queryDocument, String queryDocumentName,
-                                          GraphQLSchema graphQLSchema, AuthConfig authConfig) {
+    private SyntaxTree generateSyntaxTree(List<String> queryDocuments,
+                                          GraphQLSchema graphQLSchema, AuthConfig authConfig) throws IOException {
```
Comment (RolandHewage) [inline graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java:130]: Update the documentation comments here. Also format the method signature.

File: graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java [hunk 5]
```diff
@@ -166,11 +166,11 @@ private NodeList<ImportDeclarationNode> generateImports() {
      * @param authConfig                the object instance representing authentication configuration information
      * @return                          the node list which represent members in the client file
      */
-    private NodeList<ModuleMemberDeclarationNode> generateMembers(Document queryDocument, String queryDocumentName,
-                                                                  GraphQLSchema graphQLSchema, AuthConfig authConfig) {
+    private NodeList<ModuleMemberDeclarationNode> generateMembers(List<String> queryDocuments,
+                                              GraphQLSchema graphQLSchema, AuthConfig authConfig) throws IOException {
```
Comment (RolandHewage) [inline graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java:170]: Update the documentation comments here.

File: graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java [hunk 6]
```diff
@@ -183,12 +183,12 @@ private NodeList<ModuleMemberDeclarationNode> generateMembers(Document queryDocu
      * @param authConfig                the object instance representing authentication configuration information
      * @return                          the node which represent the client class in the client file
      */
-    private ClassDefinitionNode generateClientClass(Document queryDocument, String queryDocumentName,
-                                                    GraphQLSchema graphQLSchema, AuthConfig authConfig) {
+    private ClassDefinitionNode generateClientClass(List<String> queryDocuments,
+                                                GraphQLSchema graphQLSchema, AuthConfig authConfig) throws IOException {
```
Comment (RolandHewage) [inline graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java:187]: Update the documentation comments here.

File: graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java [hunk 7]
```diff
@@ -249,17 +249,20 @@ private FunctionDefinitionNode generateInitFunction(AuthConfig authConfig) {
      * @param authConfig        the object instance representing authentication configuration information
      * @return                  the list of nodes which represent the remote functions
      */
-    private List<FunctionDefinitionNode> generateRemoteFunctions(Document queryDocument, GraphQLSchema graphQLSchema,
-                                                                 AuthConfig authConfig) {
+    private List<FunctionDefinitionNode> generateRemoteFunctions(List<String> queryDocuments,
+                                             GraphQLSchema graphQLSchema, AuthConfig authConfig) throws IOException {
```
Comment (RolandHewage) [inline graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ClientGenerator.java:253]: Update the documentation comments here. Also reformat the method signature.

**Review by RolandHewage — COMMENTED** [review]
Added few comments. Implementation wise the approach is fine. After fixing the suggested changes this can be merged. Please update the GraphQL tool guide in the Ballerina learn page too.

File: graphql-cli/src/main/java/io/ballerina/graphql/generator/CodeGenerator.java [hunk 8]
```diff
@@ -36,15 +34,12 @@
...
-import java.io.File;
 import java.io.IOException;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
 
-import static io.ballerina.graphql.generator.CodeGeneratorConstants.CONFIG_TYPES_FILE_NAME;
-import static io.ballerina.graphql.generator.CodeGeneratorConstants.TYPES_FILE_NAME;
-import static io.ballerina.graphql.generator.CodeGeneratorConstants.UTILS_FILE_NAME;
+import static io.ballerina.graphql.generator.CodeGeneratorConstants.*;
```
Comment (RolandHewage) [inline graphql-cli/src/main/java/io/ballerina/graphql/generator/CodeGenerator.java:42]: Let's import each constant separately. Remove '*'

**Review by RolandHewage — APPROVED** [review]

**Pull request merged**